nixos/modules/services/security/bitwaren_rs/default.nix

```diff
@@ -0,0 +1,462 @@
+  }
+  // optionalAttrs (packageVersionAtLeast "1.6")
+  {
+    iconCacheTTL = mkOption {
+      type = nullOr int;
+      default = null;
+      example = 2592000;
+      description = ''
+        The cache time-to-live (TTL) for successfully obtained icons (in seconds, 0 is forever).
+      '';
+    };
+
+    iconCacheNEGTTL = mkOption {
+      type = nullOr int;
+      default = null;
+      example = 259200;
+      description = ''
+        The cache time-to-live (TTL) for unsuccessfully obtained icons (in seconds, 0 is forever).
+      '';
+    };
+
+    adminToken = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The authentication token used by the admin page.
+      '';
+    };
+  }
+  // optionalAttrs (packageVersionAtLeast "1.7")
+  {
+    templateFolder = mkOption {
+      type = nullOr str;
+      default = null;
+      example = "/path/to/templates";
+      description = ''
+        The folder path where the templates (e.g. for the emails) will be stored.
+      '';
+    };
+
+    reloadTemplates = mkEnableOption "reloading templates for every request (slow, meant for development)";
+
+    disableIconDownload = mkEnableOption "icon downloading";
+  }
+  // customOptions);
+
+  config = mkIf cfg.enable {
+    services.bitwarden_rs.webVaultFolder = mkIf cfg.enableWebVault (mkDefault "${pkgs.bitwarden_rs-vault}/share/bitwarden_rs/vault");
+
+    environment.systemPackages = [ pkgs.bitwarden_rs ];
+
+    systemd.services.bitwarden_rs = {
+      inherit description;
+      after = [ "network.target" ];
+      path = with pkgs; [ openssl ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        DynamicUser = mkIf (cfg.dataFolder == "/var/lib/bitwarden_rs") "true";
+        EnvironmentFile = configFile;
+        ExecStart = "${pkgs.bitwarden_rs}/bin/bitwarden_rs";
+        LimitNOFILE = "1048576";
+        LimitNPROC = "64";
+        PrivateTmp = "true";
+        PrivateDevices = "true";
+        ProtectHome = "true";
+        ProtectSystem = "strict";
+        WorkingDirectory = cfg.dataFolder;
+        ReadWriteDirectories = cfg.dataFolder;
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        StateDirectory = mkIf (cfg.dataFolder == "/var/lib/bitwarden_rs") "bitwarden_rs";
+      };
+      wantedBy = [ "multi-user.target" ];
+      restartTriggers = [ configFile ];
+    };
+
+    systemd.services.backup-bitwarden_rs = mkIf (cfg.backupFolder != null) {
+      description = "Backup ${description}";
+      environment = {
+        DATA_FOLDER = cfg.dataFolder;
+        BACKUP_FOLDER = cfg.backupFolder;
+      };
+      path = with pkgs; [ sqlite ];
+      serviceConfig = {
+        SyslogIdentifier = "backup-bitwarden_rs";
+        User = cfg.backupUser;
+        Group = cfg.backupGroup;
+        ExecStart = "${pkgs.bash}/bin/bash ${./backup.sh}";
```

Comment:
this service needs also to be a dynamic user when the main one is, otherwise it won't have access to /var/lib/bitwarden_rs